File: Backend/src/shared/contracts/model-metadata.ts
/**
 * ModelMetadata contract — Phase 2, Task 2.1.
 *
 * Mirrors CONTRACTS.md §43 and Orvex/ML/schemas/model_metadata.py. Every
 * AI-generated contract object carries this so results are reproducible and
 * auditable. Consumer-side: `.passthrough()` for forward-compat.
 */
import { z } from "zod";
import { isoTimestamp } from "./primitives.js";

export const ModelMetadata = z
  .object({
    provider: z.string(),
    model: z.string(),
    model_version: z.string(),
    configuration_version: z.string(),
    prompt_version: z.string(),
    temperature: z.number().default(0),
    timestamp: isoTimestamp,

    // Optional expansion fields (CONTRACTS.md §43) — kept optional for fwd-compat.
    system_prompt_hash: z.string().nullish(),
    toolset_version: z.string().nullish(),
    retrieval_version: z.string().nullish(),
    classifier_version: z.string().nullish(),
    embedding_model_version: z.string().nullish(),
  })
  .passthrough();
export type ModelMetadata = z.infer<typeof ModelMetadata>;

File: ML/schemas/model_metadata.py
"""ModelMetadata schema — mirrors CONTRACTS.md §43.

Every AI-generated contract object carries this so results are reproducible and
auditable. Field names/shape match the canonical JSON exactly (Requirement 10.1).
"""

from __future__ import annotations

from pydantic import BaseModel


class ModelMetadata(BaseModel):
    provider: str
    model: str
    model_version: str
    configuration_version: str
    prompt_version: str
    temperature: float = 0.0
    timestamp: str

    # Optional expansion fields noted in CONTRACTS.md §43 (kept optional for fwd-compat).
    system_prompt_hash: str | None = None
    toolset_version: str | None = None
    retrieval_version: str | None = None
    classifier_version: str | None = None
    embedding_model_version: str | None = None
    
